### PR TITLE
Added Password Reset API

### PR DESCRIPTION
**Pull request by iamvivekkaushik**
# Hey, I just made a Pull Request!

## Description
Added a dedicated API for changing user password.

## Motivation and Context
Almost every project requires users to be able to reset their forgotten password. This module previously didn't contain any API to handle that.

## Have you tested this? If so, how?
I ran `drf-user` with these changes over some code and it works for me.

## Checklist for PR author(s)
- [x] Changes are covered by unit tests (no major decrease in code coverage %).
- [x] All tests pass.
- [ ] Docstring coverage is **100%** via `interrogate drf_user` (I mean, we _should_ set a good example :smile:).
- [ ] Updates to documentation:
    - [ ] Document any relevant additions/changes in `README.md`.
    - [ ] Manually update **both** the `README.md` _and_ `docs/index.rst` for any new changes.
    - [ ] Any changed/added classes/methods/functions have appropriate `versionadded`, `versionchanged`, or `deprecated` [directives](http://www.sphinx-doc.org/en/stable/markup/para.html#directive-versionadded).  Find the appropriate next version in the project's [``__init__.py``](https://github.com/101loop/drf-user/blob/master/drf_user/__init__.py) file.

### CODE REVIEW

**Comment by sumit4613** [comment]
@iamvivekkaushik Thanks for raising this PR.
So we already have an account API to update user details, using that we can reset a user's password or any other detail. 

Can you please explain why your changes are required?

**Comment by iamvivekkaushik** [comment]
@sumit4613 Account API requires user to be logged in. This one on the other hand can reset when the user doesn't remember their old password.

**Comment by sumit4613** [comment]
@iamvivekkaushik Cool. :100: 
Can you write few test cases for your implementation.

**Comment by iamvivekkaushik** [comment]
@sumit4613 Yes, I can.

**Comment by sumit4613** [comment]
@iamvivekkaushik Also, please make sure your code is formatted by black. 
I think you're not using `pre-commit`, then please use the following command before committing your code **make check**

**Comment by iamvivekkaushik** [comment]
@sumit4613  Your restyled bot is still showing styling errors. I've run `make check`.

**Comment by sumit4613** [comment]
> @sumit4613 Your restyled bot is still showing styling errors. I've run `make check`.

Yep, I have seen that. I think the reason behind this is different versions of `black`, we are using the latest version of black, maybe Restyled.io not or vice versa. 
That's why I have closed the PR by restyled.io

File: drf_user/serializers.py [hunk 1]
```diff
@@ -297,3 +297,65 @@ def validate(self, attrs: dict) -> dict:
...
         return attrs
+
+
+class PasswordResetSerializer(serializers.Serializer):
+    """
+    This serializer is for password reset API.
+    otp: OTP received on your email/mobile
+    email: Email of the user whose password you're trying to reset
+    password: new password for the user
+    """
```
Comment (sumit4613) [inline drf_user/serializers.py:308]: Can you please change the docstring format to NumpyDoc, you can take references from the Contributing guide.

File: drf_user/serializers.py [hunk 2]
```diff
@@ -297,3 +297,65 @@ def validate(self, attrs: dict) -> dict:
...
+        Provides current user on the basis of property and destination
+        provided.
+        Parameters
+        ----------
+        destination: str
+            Provides value of property
+        Returns
+        -------
+        user: User
+        """
```
Comment (sumit4613) [inline drf_user/serializers.py:325]: Please restructure your docstrings like this. 

```
        """Provides current user on the basis of property and destination provided.

        Parameters
        ----------
        destination: str
            Provides value of property
        """
```

**Review by sumit4613 — CHANGES_REQUESTED** [review]
@iamvivekkaushik Thanks for working on this. 
Just minor changes are required then I can merge this to master. 
Could you please update docstrings according to NumpyDoc format then you're good to go.

Thanks!

**Review by sumit4613 — APPROVED** [review]
LGTM :100:

**Comment by iamhssingh** [comment]
The reset password is cool. As an alternative, the OTP feature was there in the system. I totally forgot to recognize that some systems prefer to go the reset password way. Thanks @iamvivekkaushik

**Comment by iamvivekkaushik** [comment]
I also need a feature to be able to add `Profile Image` for a user. Probably will add that one too.

**Comment by sumit4613** [comment]
@iamvivekkaushik Maybe you can add a new model like `userprofile` in your project to store profile image with some other details and make it's relation OneToOne with user model.

**Comment by iamvivekkaushik** [comment]
@sumit4613 sure I can, but the point of creating this as a library was that we don't have to write the same code over and over again. If I create a new model I'll have to do that again for a different project. But if I add it to this library then I won't ever have to write that piece of code again. That's just my 2 cents.

**Comment by sumit4613** [comment]
> @sumit4613 sure I can, but the point of creating this as a library was that we don't have to write the same code over and over again. If I create a new model I'll have to do that again for a different project. But if I add it to this library then I won't ever have to write that piece of code again. That's just my 2 cents.

Well yeah that's the point of creating any package/library. Would love to review your implementation of this, maybe I can learn few tricks by reviewing your code. 😄

**Pull request merged**